airflow-core/src/airflow/api_fastapi/execution_api/routes/variables.py

```diff
@@ -86,3 +90,17 @@ def put_variable(variable_key: str, body: VariablePostBody):
     """Set an Airflow Variable."""
     Variable.set(key=variable_key, value=body.value, description=body.description)
     return {"message": "Variable successfully set"}
+
+
+@router.delete(
+    "/{variable_key}",
+    status_code=status.HTTP_200_OK,
+    responses={
+        status.HTTP_401_UNAUTHORIZED: {"description": "Unauthorized"},
+        status.HTTP_403_FORBIDDEN: {"description": "Task does not have access to the variable"},
+    },
+)
+def delete_variable(variable_key: str) -> VariableDeleteBody:
+    """Delete an Airflow Variable."""
+    delete_count = Variable.delete(key=variable_key)
+    return VariableDeleteBody(delete_count=delete_count)
```

Comment:
I realise that this will be 0 or 1 always, but better to be explicit.

Reply:
Good to have a structured response than 1 or 0

Reply:
Although in theory, you could just sent True or False too since it is 0/1.

I am also ok if we don't want to do that and instead raise an error in API Server when count is 0.

So we have different options

Reply:
this should be a 204 no content response I reckon

Reply:
204 no content should be the right approach here but I want to send across the rows that were deleted, which will always be a 0 or 1. If i do a 204 no content, that will not allow me to send a response.